src/features/ProfileView/Profile.History.js

```diff
@@ -44,26 +41,56 @@ const ProfileHistory = () => {
     {restauraunt: 'Taqueria El Rey', hostStatus: 'Host', bill: 8.49}
   ];
 
+  const [biteHistory, setBiteHistory] = useState(mockBites);
+  const { state: { userId }, dispatch } = useContext(BiteShareContext);
+
   const renderBite = ({item, index}) => (<PreviousBite meal={item} index={index}/>);
 
-  //TODO
-  //Query previous bites from db
-  //Restauraunt Name - Guest / Host Status - Price
-  //Place in array in state to be rendered via Flatlist
+  useEffect(() => {
+    getAllDocuments(`users/${userId}/transactions`)
+      .then((user) => {
+        let cleanedTransactions = biteHistory;
+
+        user.forEach((session) => {
+          readASingleDocument(`users/${userId}/transactions`, session.id)
+            .then((summary) => {
+              summary = summary.data();
+              let bite = {
+                restauraunt: summary.restaurantName,
+                bill: summary.individualBills,
+                hostStatus: summary.role || 'guest'
+              };
+              cleanedTransactions.push(bite);
```

Comment:
maybe we can change `push` to `unshift`? just to render recent transactions  on top